2019/05/5.rb

```diff
@@ -0,0 +1,169 @@
+POSITIONMODE = "0"
+IMMEDIATEMODE = "1"
+
+def get_first_parameter_value(command, intcode, index)
+  first_input_index = Integer(intcode[index+1])
+  first_param_index_mode = command[-3]
+
+  if first_param_index_mode == IMMEDIATEMODE
+    first_input_value = first_input_index
+  else
+	first_input_value = Integer(intcode[first_input_index])
+  end
+  
+  return first_input_value
```

Comment:
```suggestion
  first_input_value
```

You do not need to (aka *should not*) write `return` at the end of ruby methods. The last thing that was evaluated **is** the return value.

Even crazier: You don't even need that variable ([Lint/UselessAssignment](https://docs.rubocop.org/en/stable/cops_lint/#lintuselessassignment))

```ruby
  if first_param_index_mode == IMMEDIATEMODE
    first_input_index
  else
    Integer(intcode[first_input_index])
  end
end
# That is all
```

I suggest that you install `rubocop` right away and at least autocorrect your code for the Lint, Layout and Style cops (`rubocop --auto-correct only Layout; rubocop -a only Style; rubocop -a only Lint`). I think it's better to do that right away while learning ruby instead of being annoyed by it later on

Reply:
Another good resource is also https://rubystyle.guide